app/models/manageiq/providers/autosde/storage_manager/storage_service.rb

```diff
@@ -46,16 +47,32 @@ def raw_delete_storage_service
   end
 
   def raw_update_storage_service(options = {})
-    capability_value_list = options.slice(*ext_management_system.capabilities.keys).values
-    capability_value_list.delete("-1")
+    # send resources only if edited
+    current_resources = storage_resources.ids
+    selected_resources = options["storage_resource_id"].to_a.pluck("value").map(&:to_i)
+    resources = if current_resources.sort != selected_resources.sort
+                  ext_management_system.storage_resources.find(options["storage_resource_id"].to_a.pluck("value")).pluck(:ems_ref)
+                end
+    # send capability_value_list only if edited
+    selected_capabilities_refs = options.slice(*ext_management_system.capabilities.keys).values
+    selected_capabilities_refs.delete("-1")  # get rid of the N/A option
+    current_capabilities = ext_management_system.capabilities.slice(*capabilities.keys)
+    current_capabilities_refs = []
+    capabilities.each do |key, value|
+      current_capabilities[key].each do |cap|
+        current_capabilities_refs.push(cap["uuid"]) if cap["value"] == value
+      end
+    end
+    capability_value_list = if current_capabilities_refs.sort != selected_capabilities_refs.sort
+                              selected_capabilities_refs
+                            end
 
     update_details = ext_management_system.autosde_client.ServiceUpdate(
       :name               => options['name'],
       :description        => options['description'],
-      :resources          => ext_management_system.storage_resources.find(options["storage_resource_id"].to_a.pluck("value")).pluck(:ems_ref),
-      :capability_id_list => capability_value_list
+      :resources          => resources || nil,
+      :capability_id_list => capability_value_list || nil
```

Comment:
`resources || nil` doesn't make sense (unless resources could be `false`).  If `resources` is `nil` then `nil || nil` is not useful.